be/src/storage/lake/persistent_index_sstable.cpp

```diff
@@ -32,44 +32,47 @@ Status PersistentIndexSstable::init(std::unique_ptr<RandomAccessFile> rf, const
     }
     options.block_cache = cache;
     sstable::Table* table;
-    RETURN_IF_ERROR(sstable::Table::Open(options, rf.get(), sstable_pb.filesize(), &table));
+    RETURN_IF_ERROR(sstable::Table::Open(options, rf.get(), sstable_pb.filesize(), sstable_pb.first_key(),
+                                         sstable_pb.last_key(), &table));
     _sst.reset(table);
     _rf = std::move(rf);
     _sstable_pb.CopyFrom(sstable_pb);
     return Status::OK();
 }
 
-Status PersistentIndexSstable::build_sstable(
-        const phmap::btree_map<std::string, std::list<IndexValueWithVer>, std::less<>>& map, WritableFile* wf,
-        uint64_t* filesz) {
+Status PersistentIndexSstable::build_sstable(const phmap::btree_map<std::string, IndexValueWithVer, std::less<>>& map,
+                                             WritableFile* wf, uint64_t* filesz, FirstLastKeys* keys) {
     std::unique_ptr<sstable::FilterPolicy> filter_policy;
     filter_policy.reset(const_cast<sstable::FilterPolicy*>(sstable::NewBloomFilterPolicy(10)));
     sstable::Options options;
     options.filter_policy = filter_policy.get();
     sstable::TableBuilder builder(options, wf);
-    for (const auto& [k, v] : map) {
+    for (const auto& [k, index_value_with_ver] : map) {
         IndexValuesWithVerPB index_value_pb;
-        for (const auto& index_value_with_ver : v) {
-            auto* value = index_value_pb.add_values();
-            value->set_version(index_value_with_ver.first);
-            value->set_rssid(index_value_with_ver.second.get_rssid());
-            value->set_rowid(index_value_with_ver.second.get_rowid());
-        }
+        auto* value = index_value_pb.add_values();
+        value->set_version(index_value_with_ver.first);
+        value->set_rssid(index_value_with_ver.second.get_rssid());
+        value->set_rowid(index_value_with_ver.second.get_rowid());
         builder.Add(Slice(k), Slice(index_value_pb.SerializeAsString()));
     }
     RETURN_IF_ERROR(builder.Finish());
     *filesz = builder.FileSize();
+    *keys = std::move(builder.FirstLastKeys());
     return Status::OK();
 }
 
 Status PersistentIndexSstable::multi_get(const Slice* keys, const KeyIndexSet& key_indexes, int64_t version,
                                          IndexValue* values, KeyIndexSet* found_key_indexes) const {
     std::vector<std::string> index_value_with_vers(key_indexes.size());
     sstable::ReadOptions options;
+    sstable::ReadIOStat stat;
```

Comment:
swap line 67 and line 68, ensure that `stat` lives longer than `options` so the `options` won't hold an invalid reference of `stat`